### PR TITLE
[FSDP][Easy] nit follow-ups to handle refactor

### PR DESCRIPTION
**Pull request by awgu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #105738

### CODE REVIEW

File: torch/distributed/fsdp/_exec_order_utils.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def init(
             index = len(self.all_handles)
             self.all_handles.append(handle)
             handle._handle_index = index
-            self.param_to_fqn = _get_param_to_fqns(root_module)
```
Comment (awgu) [inline torch/distributed/fsdp/_exec_order_utils.py:82]: This does not need to be in a loop `for handle in traversal_utils._get_fsdp_handles(root_module):`. The previous PR accidentally indented this.

**Review by fegin — APPROVED** [review]

**Comment by awgu** [comment]
Failing tests are the known unstable DTensor ones:
[linux-bionic-cuda11.8-py3.10-gcc9 / test (distributed, 3, 3, linux.8xlarge.nvidia.gpu, unstable)](https://hud.pytorch.org/pr/pytorch/pytorch/105738#15239378499) ([gh](https://github.com/pytorch/pytorch/actions/runs/5623400655/job/15239378499))
[linux-focal-py3.8-gcc7 / test (distributed, 1, 2, linux.2xlarge, unstable)](https://hud.pytorch.org/pr/pytorch/pytorch/105738#15238373994) ([gh](https://github.com/pytorch/pytorch/actions/runs/5623400655/job/15238373994))

**Review by voznesenskym — APPROVED** [review]

**Pull request closed without merging**